spec/features/forms/form/item_edit_spec.rb

```diff
@@ -6,68 +6,105 @@
   INDENT = 40
 
   let!(:user) { create(:user) }
-  let!(:form) do
-    create(:form, name: "Foo", question_types: %w[integer multilevel_select_one select_one integer])
-  end
   let(:forms_path) { "/en/m/#{form.mission.compact_name}/forms" }
   let(:qings) { form.root_group.c }
+  let!(:form) { create(:form, name: "Foo", question_types: question_types) }
+  let(:edit_url) { edit_form_path(form, locale: "en", mode: "m", mission_name: get_mission.compact_name) }
 
   before do
     login(user)
   end
 
-  scenario "add nested groups to a form" do
-    outer_group = "Outer Group"
-    inner_group = "Inner Group"
-    question_name = form.questions.first.name
-
-    visit(edit_form_path(form, locale: "en", mode: "m", mission_name: get_mission.compact_name))
-    create_group(outer_group)
-    create_group(inner_group)
+  context "with no groups" do
+    let(:question_types) { %w[integer multilevel_select_one select_one integer] }
+
+    scenario "adding nested groups" do
+      outer_group = "Outer Group"
+      inner_group = "Inner Group"
+      question_name = form.questions.first.name
+
+      visit(edit_url)
+      create_group(outer_group)
+      create_group(inner_group)
+
+      within(".form-items .draggable-list-wrapper") do
+        expect(page).to have_nested_item(path: [4], name: outer_group)
+        expect(page).to have_nested_item(path: [5], name: inner_group)
+
+        # nest inner group under outer group
+        drag_item(from: [5], to: [5], indent: true)
+        expect(page).to have_nested_item(path: [4, 0], name: inner_group)
+
+        # nest first question under outer group
+        drag_item(from: [0], to: [4, 0])
+        expect(page).to have_nested_item(path: [3, 1], name: question_name)
+      end
+
+      # Edit a group
+      form_item([3, 0]).find(".fa-pencil").click
+      expect(page).to have_css(".modal-title", text: "Edit Group")
+      fill_in("Name (English)", with: "Inner Groupe")
```

Comment:
why spelling group as Groupe?

Reply:
Because the original value was 'Group' and we are testing that it gets changed.